packages/gatsby-source-wordpress/src/normalize.js

```diff
@@ -192,15 +192,17 @@ exports.mapPostsToTagsCategories = entities => {
   const categories = entities.filter(e => e.__type === `wordpress__CATEGORY`)
 
   return entities.map(e => {
-    if (e.__type === `wordpress__POST`) {
+    let hasCategories = (e.categories && e.categories.length);
+    let hasTags = (e.tags && e.tags.length);
+    if (e.__type === `wordpress__POST` || hasCategories || hasTags ) {
```

Comment:
this check is not needed - you do check for `hasCategories`/`hasTags` below anyway

IMO this could be done with (lines refer to current master branch - not your PR):
 - removing ``if (e.__type === `wordpress__POST`) {`` check in line 195
 - replacing `if (e.tags.length) {` with `if (e.tags && Array.isArray(e.tags) && e.tags.length) {` in line 197 (and same for categories in line 203
   (added isArray check because we no longer check this just for posts that we could assume would be array)

Reply:
Thank you for your review, you're right I dont need the `wordpress__POST` check. I just did an another commit